jsonschema/_validators.py

```diff
@@ -190,9 +190,14 @@ def enum(validator, enums, instance, schema):
 
 
 def ref(validator, ref, instance, schema):
-    with validator.resolver.resolving(ref) as resolved:
+    scope, resolved = validator.resolver.resolve(ref)
+    validator.resolver.push_scope(scope)
```

Comment:
Got it. OK, I'd assume there's some cost in generator-nonsense with `contextmanager`, but I'm not sure I'd have guessed it'd be insignificant to switch to the regular protocol. Regardless, this way is fine with me, but if I have an overabundance of time (ha) I'll double check.